File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
    std::istringstream iss(s);
    std::vector<std::string> words;
    std::string word;

    // Split the string into words
    while (iss >> word) {
        words.push_back(word);
    }

    // Use a string stream to build the reversed string
    std::ostringstream oss;

    // Reverse loop to append words in reverse order
    for (int i = words.size() - 1; i >= 0; i--) {
        oss << words[i];
        if (i > 0) oss << ' ';
    }

    return oss.str();
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        //break down s into different words
        String[] words = s.split(" +");
        
        //StringBuilder
        StringBuilder sb = new StringBuilder();
        
        //run an opposite loop
        for(int i = words.length - 1; i >= 0; i--){
            sb.append(words[i]);
            sb.append(" ");
        }
        
        return sb.toString().trim();
    }
}